### PR TITLE
chore(compactor2): Assign all rows and sizes in simulator

### PR DESCRIPTION
**Pull request by alamb**
~Draft as it builds on https://github.com/influxdata/influxdb_iox/pull/7123 and will conflict with https://github.com/influxdata/influxdb_iox/pull/7079 which is a higher priority~

The current simulator can "lose rows" when compacting parquet files due to rounding error. This PR fixes that by adjusting the last file created with the delta

### CODE REVIEW

File: compactor2/tests/layouts/core.rs [hunk 1]
```diff
@@ -94,17 +94,17 @@ async fn all_overlapping_l0() {
     - "L0.3[100,200]       |-------------------------------------L0.3-------------------------------------|"
     - "L0.2[100,200]       |-------------------------------------L0.2-------------------------------------|"
     - "L0.1[100,200]       |-------------------------------------L0.1-------------------------------------|"
-    - "**** 2 Output Files (parquet_file_id not yet assigned), 89.1mb total:"
+    - "**** 2 Output Files (parquet_file_id not yet assigned), 90mb total:"
```
Comment (alamb) [inline compactor2/tests/layouts/core.rs:97]: it is nice to see the same output size as input size 😅

File: compactor2/tests/layouts/large_overlaps.rs [hunk 2]
```diff
@@ -871,75 +871,74 @@ async fn many_good_size_l0_files() {
...
     - "**** Final Output Files "
     - "L2                                                                                                  "
     - "L2.289[0,50] 100mb  |--L2.289---|                                                                   "
-    - "L2.300[51,102] 101.2mb              |---L2.300---|                                                    "
-    - "L2.303[103,155] 101.12mb                            |---L2.303---|                                      "
-    - "L2.304[156,207] 99.17mb                                           |---L2.304---|                       "
-    - "L2.305[208,230] 42.78mb                                                         |L2.305|               "
-    - "L2.306[231,276] 86.8mb                                                                |--L2.306--|    "
-    - "L2.307[277,288] 21.22mb                                                                            |L2.307|"
+    - "L2.297[179,229] 100.92mb                                                 |--L2.297---|                  "
```
Comment (alamb) [inline compactor2/tests/layouts/large_overlaps.rs:936]: It is interesting this output is slightly different -- is it OK @NGA-TRAN ?

**Review by Joe-Blount — APPROVED** [review]
LGTM

File: compactor2/tests/layouts/core.rs [hunk 3]
```diff
@@ -159,17 +159,28 @@ async fn all_non_overlapping_l0() {
...
     - "  Soft Deleting 10 files: L0.1, L0.2, L0.3, L0.4, L0.5, L0.6, L0.7, L0.8, L0.9, L0.10"
     - "  Creating 2 files at level CompactionLevel::L1"
-    - "**** Final Output Files "
+    - "**** Simulation run 1, type=split(split_times=[720]). 2 Input Files, 100mb total:"
     - "L1                                                                                                  "
+    - "L1.12[721,901] 20.09mb                                                                |----L1.12----| "
     - "L1.11[0,720] 79.91mb|----------------------------L1.11----------------------------|                 "
-    - "L1.12[721,901] 19.98mb                                                                |----L1.12----| "
+    - "**** 2 Output Files (parquet_file_id not yet assigned), 100mb total:"
+    - "L2                                                                                                  "
```
[the remark this answers is not in the document]
Comment (NGA-TRAN) [reply compactor2/tests/layouts/core.rs:174]: Each file needs >= 100MB to get upgraded. In this case 2 L1s are compacted and then resplit into 2 L2s. I think the setup is 80% and 20% split ratio

**Review by NGA-TRAN — APPROVED** [review]
LGTM

**Pull request merged**